src/section/section_keys.rs

```diff
@@ -26,13 +26,17 @@ pub struct SectionKeysProvider {
     /// A cache for current and previous section BLS keys.
     cache: MiniKeyCache,
     /// The new keys to use when section update completes.
-    pending: Option<SectionKeyShare>,
+    // TODO: evict outdated keys.
+    // TODO: alternatively, store the pending keys in DkgVoter instead. That way the outdated ones
+    //       would get dropped when the DKG session itself gets dropped which we already have
+    //       implemented.
+    pending: HashMap<bls::PublicKey, SectionKeyShare>,
```

Comment:
or use LRU cache to get outdated keys faded away?

Reply:
Yep, we could do that, but I usually try to avoid LRU cache if possible because it's not precise. You either end up keeping stuff in the cache for too long or you remove it too soon. Better to remove things when you are sure you don't need them, not sooner, not later, if possible.